ext/spl/spl_array.c

```diff
@@ -1156,22 +1187,23 @@ static const zend_object_iterator_funcs spl_array_it_funcs = {
 
 zend_object_iterator *spl_array_get_iterator(zend_class_entry *ce, zval *object, int by_ref) /* {{{ */
 {
-	zend_user_iterator *iterator;
+	spl_array_iterator *iterator;
 	spl_array_object *array_object = Z_SPLARRAY_P(object);
 
 	if (by_ref && (array_object->ar_flags & SPL_ARRAY_OVERLOADED_CURRENT)) {
 		zend_throw_error(NULL, "An iterator cannot be used with foreach by reference");
 		return NULL;
 	}
 
-	iterator = emalloc(sizeof(zend_user_iterator));
+	iterator = emalloc(sizeof(spl_array_iterator));
```

Comment:
Not sure what the style for PHP is, butin other projects the following is preferred:

```suggestion
	iterator = emalloc(sizeof(*iterator));
```